ui/webui/src/app.jsx

```diff
@@ -33,15 +41,60 @@ export const Application = () => {
 
     useEffect(() => cockpit.file("/run/anaconda/bus.address").watch(setAddress), []);
 
+    const steps = [
+        {
+            id: 0,
+            name: "Installation language",
+            component: <AddressContext.Provider value={address}>
+                <InstallationLanguage />
+            </AddressContext.Provider>
+        },
+        {
+            id: 1,
+            name: "Storage configuration",
+            component: <AddressContext.Provider value={address}>
```

Comment:
Please avoid code duplication and use this as a wrapper function. 

I will write the diffs for start, because I think JS syntax can be intimidating, but let me know if you prefer to let you explore it all the way

```
+    const wrapWithContext = children => {
+        return (
+            <AddressContext.Provider value={address}>
+                {children}
+            </AddressContext.Provider>
+        );
+    };
+
     const steps = [
         {
             id: 0,
             name: "Installation language",
-            component: <AddressContext.Provider value={address}>
-                <InstallationLanguage />
-            </AddressContext.Provider>
+            component: wrapWithContext(<InstallationLanguage />)
         },

....
```

Reply:
Much better, thanks, implemented! :)